postgres/tasks_remote.py

```diff
@@ -53,34 +58,87 @@ def setup(ctx):
-        ctx.local(f"sed -i 's/OWNER TO surf/OWNER TO postgres/g' {snapshot_file_path}", echo=True)
+        conn.local(f"sed -i 's/OWNER TO surf/OWNER TO postgres/g' {snapshot_file_path}", echo=True)
 
-    # Recreating database with all correct privileges
-    if recreate:
-        print("Recreating databases")
-        setup(ctx)
+    if "test" in conn.config.surfconext.oidc_endpoint:
+        print("Rewriting team ARN's for test communities")
+        production_test_team_urn = "urn:collab:group:surfteams.nl:nl:surfnet:diensten:test_team_zoekportal"
+        development_test_team_urn = "urn:collab:group:test.surfconext.nl:nl:surfnet:diensten:zoekportaal_test_community"
+        conn.local(f"sed -i 's/{production_test_team_urn}/{development_test_team_urn}/g' {snapshot_file_path}",
+                   echo=True)
 
     print("Restoring snapshot")
     # Setup auto-responder
-    postgres_user = ctx.config.django.postgres_user
-    postgres_password = ctx.config.secrets.postgres.password
+    postgres_user = conn.config.django.postgres_user
+    postgres_password = conn.config.secrets.postgres.password
     postgres_password_responder = Responder(pattern=r"Password: ", response=postgres_password + "\n")
     # Run Postgres command with port forwarding
-    with ctx.forward_local(local_port=1111, remote_host=ctx.config.django.postgres_host, remote_port=5432):
-        ctx.local(f"psql -h localhost -p 1111 -U {postgres_user} -W -d edushare -f {snapshot_file_path}",
-                  echo=True, watchers=[postgres_password_responder], pty=True)
+    with conn.forward_local(local_port=1111, remote_host=conn.config.django.postgres_host, remote_port=5432):
+        conn.local(f"psql -h localhost -p 1111 -U {postgres_user} -W -d edushare -f {snapshot_file_path}",
+                   echo=True, watchers=[postgres_password_responder], pty=True)
+
+    if add_superuser:
+        print("Adding superuser")
+        password = conn.config.secrets.postgres_application.password
+        insert_user = insert_django_user_statement("supersurf", password, is_edushare=True)
+        with conn.forward_local(local_port=1111, remote_host=conn.config.django.postgres_host, remote_port=5432):
+            conn.local(
+                f'psql -h localhost -p 1111 -U {postgres_user} -d edushare -W -c "{insert_user}"',
+                echo=True, warn=True, watchers=[postgres_password_responder], pty=True
+            )
+
+    print("Done")
+
+
+@task()
+def create_snapshot(conn):
+    """
+    Dumps database content as SQL creating a snapshot that other nodes can load easily
+    """
+    if conn.host != conn.config.aws.bastion:
+        raise Exit("Did not expect the host {conn.host} while the bastion is {conn.config.aws.bastion}")
+
+    today = date.today()
+    dump_file = f"edushare.{today:%Y-%m-%d}.postgres.sql"
+    dump_file_path = os.path.join("postgres", "dumps", dump_file)
+    print("Creating snapshot:", dump_file)
+
+    excluded_tables = [
```

Comment:
Je hebt het vast getest, maar er zit nu geen data in die foreign keys mist?

Reply:
Nee